pkg/apis/core/v1beta1/defaults_shoot.go

```diff
@@ -295,6 +296,21 @@ func SetDefaults_Worker(obj *Worker) {
 	if obj.UpdateStrategy == nil {
 		obj.UpdateStrategy = ptr.To(AutoRollingUpdate)
 	}
+
+	if obj.UpdateStrategy != nil && (*obj.UpdateStrategy == AutoInPlaceUpdate || *obj.UpdateStrategy == ManualInPlaceUpdate) {
```

Comment:
Why not using https://github.com/gardener/gardener/blob/9f78a779c44ae2831ea2459b12a7c1df499ca613/pkg/apis/core/v1beta1/helper/shoot.go#L662-L668 ?

Reply:
No need to check nil because, we set the field just above it